### ISSUE TITLE
Spring Boot 1.5.1 released

### ISSUE THREAD

**Issue by cbornet** [jhipster/generator-jhipster#5045]
https://spring.io/blog/2017/01/30/spring-boot-1-5-1-released

**Comment by cbornet** [comment]
with Kafka autoconfiguration

**Comment by jdubois** [comment]
Yes, but we'll do it after the 4.0 release, as it might break stuff.

- For Kafka I think they just included Spring Kafka, so it shouldn't change
much for us
- For the logger endpoint, it looks like they copied us (again !), so we'll
need to refactor and remove our code


Le 1 févr. 2017 9:08 AM, "Christophe Bornet" <notifications@github.com> a
écrit :

**Comment by cbornet** [comment]
I'm glad when they copy us : less is more 😄

**Comment by gzsombor** [comment]
It would be nice to have a more advanced log endpoint, where you could actually see the logs :) Sure, there are log servers, with all the bells and whistles - but sometimes you don't need that much, just to look into log, to know what happened 10 minutes ago :)

**Comment by jdubois** [comment]
@gzsombor  Yes, we would need some kind of in-memory data store for that. We could also have the same thing for the security audits, to improve log in performance. I would love to have some kind of simple hashmap or list, which would just store the last 10,000 events, for example.

**Comment by erikkemperman** [comment]
For what it's worth -- I'm playing around with a branch that has SB1.5 (and some other upgrades). It's actually not that big of a deal, as far as I can see so far.

Background is I was trying to update my JH app to 3.12.2 (after ~0.5 year hiatus) but ran into some difficulties, which seem to be due to slightly unaligned versions of hibernate components, resulting in a helper class (FieldInterceptionHelper) not being available. It seems this is only a problem when running from a war file, and then only with Tomcat. But unfortunately I really need precisely that combination.

I've found some issues on SB which look like they are the same thing and apparently 1.5 makes this go away.

Currently I'm having some trouble with hibernate ("missing sequence") after  generating some simple entities -- but that also seems to be happening with current master? I'll see if I can get that resolved quickly, or will make a separate ticket for it.

Anyway, if anyone is interested to go SB1.5 in the near future let me know, I'd be happy to submit what I have here.

**Comment by jdubois** [comment]
No, you should not have any "missing sequence" with master.

Le 1 févr. 2017 1:06 PM, "Erik Kemperman" <notifications@github.com> a
écrit :

**Comment by erikkemperman** [comment]
Well, I just tried again with an even simpler setup (freshly checked out master branch, all default generator choices, a single entity with a single un-validated field) and I still get the same exception.

But this is getting off-topic here, sorry about that. I will see if I can fix it and create a separate issue otherwise.

**Comment by erikkemperman** [comment]
Sorry again for off-topic, but just in case anyone runs into the same "missing sequence" problem and finds this thread -- I had done `npm link` from the generator directory, but then I did _not_ do `npm link generator-jhipster` in between `yo jhipster` and `yo jhipster:entity Foo`, which would have avoided my troubles.

So the first `yo` command would use my linked generator but the second `yo` would then prefer the one that just got installed in the local node_modules, from the npm repo, which was an incompatible version in this case. I somehow never noticed that the generated app even has its own dependency on the generator, but that kind of makes sense now that I think about it.

I wonder if it makes sense to have version `'next'` or `'dev'` or something like that in the package.json of the master branch to avoid this kind of thing? Instead of a version string that actually matches a published version of the generator that is (possibly) not compatible?

**Comment by jdubois** [comment]
@erikkemperman no that's normal, if you want to use a dev version, you need to do the "link" command in the directory.

**Comment by erikkemperman** [comment]
@jdubois Thanks, yes that is what I found out today :-) I just thought that an unpublished version string in the master branch would make this kind of mistake impossible (because then `npm install` would fail for the original `yo jhipster`). But fair enough, I will probably never get this wrong again.

**Comment by erikkemperman** [comment]
In case anyone is interested in going to spring-boot 1.5.1: I have almost everything working including some other (relatively minor) upgrades. Where possible I have aligned with the versions in [spring-boot-dependencies](https://github.com/spring-projects/spring-boot/blob/v1.5.1.RELEASE/spring-boot-dependencies/pom.xml) (and omitted explicit version strings in JH in such cases).

https://github.com/erikkemperman/generator-jhipster/tree/springboot151
https://github.com/erikkemperman/jhipster/tree/springboot151

All tests are now passing except for ng2-jwt -- but unfortunately I can't justify spending much more time on this, because it's not immediately relevant for my own project and I _really_ have to get on with that now.

So, if anyone has a quick clue what might be going wrong that would be much appreciated: [here](https://travis-ci.org/erikkemperman/generator-jhipster/builds/199228622) are my latest Travis results.

Some further remarks / questions:

- There is an [issue](https://github.com/jsevellec/cassandra-unit/issues/212) in cassandra-unit which causes the unit tests to fail in the ng2-cassandra sample, cf. . I've added a line to `cassandra-random-port.yml` as a workaround but this should be removed with the next upgrade of cassandra-unit.

- I had to forcefully pull in spring-cloud-netflix at 1.2.5.RELEASE because the transitive dependency I got otherwise (1.2.4-RELEASE) relied on a class that was deprecated in SB1.4 and now removed. This is a bit ugly but could go away pretty soon I would hope.

- The SB1.5.1 releases notes [mention](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-1.5-Release-Notes#actuator-security) that default security for some endpoints were moved from ADMIN to ACTUATOR roles. I wasn't sure of the best way to deal with this, for the moment I've simply added `management.security.roles=ADMIN` to the configuration.

- There is another  [mention](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-1.5-Release-Notes#devtools-excluded-by-default) on the SB1.5.1 release notes about devtools, that I'm not sure if/how JH is affected by it. Thoughts?

- It looks like the gatling configuration distributed with JH might have already been out of date with the last bump in `pom.xml` / `gradle.properties`. So I just copied the current defaults from upstream into `gatling.conf`... Does that make sense, or did I overlook some subtle trick here?

**Comment by PierreBesson** [comment]
@erikkemperman Please go on with the PR, it would be bad that your work go to waste as we need to migrate to SB1.5 sooner or later.

* About the spring cloud dependency it is probably because we still depends on Brixton rather than the newer Camdem. Maybe we should move to Camdem but we will need to do some testing (or put our trust in Travis 👷 ).
* About Actuator, I think your solution is OK. But I'm wondering if we could not instead add actuator role to the admin user, the actuator role could be given to several users that will be able to view our status dashboards. Then I'm not sure if it would be useful at all.
* devtools: we should probably remove the `shell` has it is going to be deprecated and few people use it.

**Comment by erikkemperman** [comment]
@PierreBesson 
Actually I already switched to Camden, but I guess some of the other spring-cloud stuff is still catching up. Anyway I expect my manual override dependency would be temporary.

I'd like to make a PR but still have 1 failing test that I wasn't able to figure out quickly..

**Comment by erikkemperman** [comment]
Oh, also I didn't look at Kafka auto config at all!

**Comment by jdubois** [comment]
Lots of work here! @erikkemperman yes please do a PR, so your work is not lost, and so you benefit from Travis - as it's not complete, just add [WIP] (for "work in progress") on the title

**Comment by erikkemperman** [comment]
All right, I might find some time tomorrow to annotate a bit and maybe make some checkboxes with todos of the work in progress. Thanks for the quick response guys!

**Comment by deepu105** [comment]
If required we can merge your PR to a local branch and work on top to fix
whats failing so that your commit credit stays as such

Thanks & regards,
Deepu

**Comment by jdubois** [comment]
Argh I broke the build at https://travis-ci.org/jhipster/generator-jhipster/builds/203462647

I don't understand the JWT issue - I don't have it on my test project, which looks a lot like the Travis build, but when I use the Travis config I do reproduce the issue...

I'm not happy with this new Spring Boot release, lots of stuff broken, and that's the first time we have so many issues with it.

**Comment by jdubois** [comment]
The JWT issue is incredible:

- It fails with Maven and Gradle (and that's why Travis fails)
- It works "most of time" in my IDE (that's why I didn't see the issue)

This looks like a build issue, and that's quite worrying

**Comment by pascalgrimaud** [comment]
I'm having a look about Travis build

**Comment by pascalgrimaud** [comment]
@jdubois :
1) for the ng2-jwt build, there is an issue in Audit page. Not related to spring-boot 1.5 IMO. Here the screenshot of the error:
![ng2-jwt-error](https://cloud.githubusercontent.com/assets/9156882/23140632/f1c8d35c-f7b2-11e6-863e-c4969a40e378.png)

2) for the 2nd failed build, it is related to cucumber:
```
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 13.217 s
[INFO] Finished at: 2017-02-21T05:25:14+11:00
[INFO] Final Memory: 47M/261M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.0:testCompile (default-testCompile) on project travis-hazelcast-cucumber: Compilation failure: Compilation failure:
[ERROR] /home/travis/app/src/test/java/io/github/jhipster/travis/cucumber/stepdefs/StepDefs.java:[5,37] cannot find symbol
[ERROR] symbol:   class SpringApplicationContextLoader
[ERROR] location: package org.springframework.boot.test
[ERROR] /home/travis/app/src/test/java/io/github/jhipster/travis/cucumber/stepdefs/StepDefs.java:[11,76] cannot find symbol
[ERROR] symbol: class SpringApplicationContextLoader
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

I don't have time tonight for trying to fix these issues, just confirmed and reported them

**Comment by erikkemperman** [comment]
@jdubois @pascalgrimaud The cucumber error was resolveer in my original PR, see https://github.com/jhipster/generator-jhipster/pull/5146/files#diff-5f91fff21ab66a85e9191a764af094ccR12

The app-ng2-jwt issue I noticed but couldn't figure out quickly. I mentioned this failing test on the PR,  so I assumed it was fixed by Julien as he merged the update.

**Comment by pascalgrimaud** [comment]
@erikkemperman : can you do a quick PR to fix cucumber ?

**Comment by erikkemperman** [comment]
@pascalgrimaud Yep, I'm on it

**Comment by erikkemperman** [comment]
https://github.com/jhipster/generator-jhipster/pull/5266

**Comment by erikkemperman** [comment]
@jdubois I noticed the companion PR on the serverside lib had merge conflicts since I submitted it, so I've rebased my branch.

**Comment by pascalgrimaud** [comment]
@erikkemperman : I answer to you here about ng2-jwt
- the Audit page didn't appear correctly because audit.data is empty (size is 0)
- this same audit.data is not empty when it's not JWT
- so maybe something has changed in spring-boot actuator, I didn't find the cause yet

**Comment by pascalgrimaud** [comment]
Just generated a fresh project with JHipster 3.12.2, AngularJS 1, Spring Boot 1.4.2. The column Extra Data in the Audit page is empty.
So it seems we have the same behavior now, after my PR #5267

**Comment by erikkemperman** [comment]
@pascalgrimaud Thanks for looking into this, it looks like currently master is building correctly even in this case: https://travis-ci.org/jhipster/generator-jhipster/builds/203611247

**Comment by erikkemperman** [comment]
@jdubois It looks like I might have some time today to go over the remaining version bumps (and deletions to align with SB implicitly) and demux into separate commits. Would you prefer a single PR with those separate commits, or also separate PRs?

**Comment by jdubois** [comment]
- For the JWT issue: I have no idea what happens, but this seems linked to old cached data in the browser. I've stopped looking at this for the moment, as it looks specific to my browser.
- @erikkemperman yes one PR per "big enough" change, as we might want to revert them one by one, and see when the Travis tests fail also

**Comment by jdubois** [comment]
Closing this as the migration looks ok

**Comment by erikkemperman** [comment]
@jdubois All right, but it will be a lot of PRs if I separate every upgrade! By the way yarnpkg is driving me crazy, I keep getting failing Travis builds and inevitably find that `yarn install` failed. I would *love* to go back to `npm` at least for the Travis config...

**Comment by jdubois** [comment]
OK, if that's easier for you, just do one big PR
I agree we have a lot of Travis issues at the moment - it was more stable with NPM, but also far slower....

**Comment by erikkemperman** [comment]
I don't mind making separate PRs but I think you guys might get a bit annoyed from so many small ones :-)

**Comment by pascalgrimaud** [comment]
@erikkemperman : I know the problem about Travis build since we migrated to yarn. I have already worked on a specific branch to reduce these random failures. I should commit it soon

**Comment by pascalgrimaud** [comment]
@jdubois : let me trying to resume what I analyzed:
- before JHipster 4.0, with AngularJS1
    - without JWT: the audit page works well
    - with JWT: the last column (about extra data) is empty. I think it comes directly from the AuditResource, in the backend
- now, JHipster 4.0+, with Angular2
    - without JWT: the audit page works well
    - with JWT: in the front, there is an error because the audit.data is empty. I resolved it in #5267, but the column will still be empty, so we have the same behavior than before

So for me, there is still 1 issue, with JWT: the last column about extra is empty
It is independant from spring-boot 1.5 and Angular2, IMO

**Comment by erikkemperman** [comment]
I'm going to make a few PRs, as the test results on my fork come in.

In the end I thought it best to make just a few PRs, but keep the commits separate. That way it's not too much spam for you guys, while we can still cherry-pick / revert small changes individually if needed.
